### DISCUSSION TITLE
Handling custom predictor storage

### DISCUSSION THREAD

**Discussion by wagnersemantix** [kubeflow/kfserving#2976] [Q&A] [answered]
First time writing a [custom predictor](https://kserve.github.io/website/master/modelserving/v1beta1/custom/custom_model/#environment-variables) and trying to understand how is data made acessible to the predictor.

My understanding is that, by specifying an compatible storage system in the `STORAGE_URI`, the whole storage is somehow mounted as a local filesystem to the container, is that correct? 
If so, is it unnecessary to create a download step to the container filesystem or should that still be implemented?

Thanks in advance

**Comment by elukey** [comment]
Hi! What happens is that a specific container called `storage-initializer` will pull the model file from the URL stated in `STORAGE_URI`. For example if you use an s3 endpoint, behind the scenes the storage-initializer will use the boto library to pull the model binary file on the pod's file system.

**Reply by wagnersemantix** [reply]
Thank you for your reply.
If my understanding is correct the `STORAGE_URI` is used for a single file. Is there a prefered way to work with predictors that require multiple files?

**Reply by elukey** [reply, marked as answer]
@wagnersemantix is supports also multiple files if you use s3, for example this is one of the configs I have in my isvcs for LLM files: 
```
      - name: STORAGE_URI
        value: s3://wmf-ml-models/bloom/bloom-3b/
```
